app/core/store/modules/campaigns.js

```diff
@@ -0,0 +1,30 @@
+import campaignsApi from 'core/api/campaigns'
+
+export default {
+  namespaced: true,
+  state: {
+    byId: {},
+    bySlug: {}
+  },
+
+  mutations: {
+    setCampaignData: (state, campaignData) => {
+      Vue.set(state.byId, campaignData._id, campaignData)
+      Vue.set(state.bySlug, campaignData.slug, campaignData)
+    }
+  },
+
+  actions: {
+    fetch: ({ commit, state }, campaignHandle) => {
```

Comment:
Nitpick: Does this fetch action method support an async function syntax?

Reply:
Yes it does, will update it